### PR TITLE
fix(module:select): Null option shows in SelectContent

### LINKED ISSUE #1587: SelectOption bug when Value = 0
Hi team!
I get a bug when using a select input and value of SelectOption = 0 

```
    <Select @bind-Value="@_selectedValue2"
            Style="width: 120px;"
            TItemValue="int"
            TItem="string">
        <SelectOptions>
            <SelectOption TItemValue="int" TItem="string" Value=@(0) Label="zero" />
            <SelectOption TItemValue="int" TItem="string" Value=@(1) Label="one" />
            <SelectOption TItemValue="int" TItem="string" Value=@(2) Label="two" />
        </SelectOptions>
    </Select>

    @code {
      private int _selectedValue2 = 0;
    }
```

Further technical details
- AntDesign Nuget Package version 0.8.1
ant-blazor-select-bug

### PR DESCRIPTION
**Pull request by anddrzejb**
### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
Fixes #1587 
Addresses issues mentioned in [this comment](https://github.com/ant-design-blazor/ant-design-blazor/pull/1996#pullrequestreview-771855643).
Addresses proposal in [this comment](https://github.com/ant-design-blazor/ant-design-blazor/pull/1906#issuecomment-913105602).

### 💡 Background and solution
To fix the primary issue, in this PR I added a boolean that identifies if there is a `SelectOptionItem` that has its `Value` equal to the default value of `TItemValue`. I use that flag to figure out what to do when the value is selected and when the `Select` is cleared. On top of that, it also adds the parameter `ValueOnClear` which will be used as a primary value when `Select` is cleared. `ValueOnClear` may be set to a value that does not exist in `SelectOptionItem`. In that case, the `SelectContent` is cleared. Otherwise it is set to the existing `SelectOptionItem`. I included test cases to cover for these scenarios (I think I cover all possibilities, at least until proven otherwise :smirk:). 

During my tests I found out we have some weird issues with `Select`. It seems it relies a lot on things happening during re-rendering. I am not a fan of this, as it splits responsibilities and is really hard to follow (and thus debug). In my opinion this makes `Select` more complex & less performant. 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  new `[Parameter] ValueOnChange` - stored `Value` that will be used when clear button is pressed      |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by anranruye** [comment]
Well, I didn't have a look at the codes yet. From your description, I feel this solution makes things more complicated. As we discussed before, the basic logic for `Select` should be simpler. With no fallback strategy, the `Select` component accepts all values of `TItemValue`, if the value is in select-options, then we display that item, otherwise we display nothing or a placeholder identities nothing(null) or a placeholder identities a not-null value which is not in the options. When clear, set the value to default(TItemValue) or ValueOnClear. 

With a fallback strategy, when the value is:
1. value in select-options, display that item, no fallback
2. null, display a placeholder, no fallback
3. fallback value defined by the strategy(null, default(TItemValue) or a user specified value), display nothing or a placeholder
4. other values, fallback

That's all. And for most cases, we don't need a fallback strategy, which makes things as simple as I mentioned in the first paragraph.

**Comment by jeffraska** [comment]
Hi, I think this PR breaks Two-way binding in Default mode while changing binded value to NULL.

- When I dont have any special NULL value option in DataSource:
1. `Select.OnValueChange()` with NULL value is called.
2. TItemValue is `long?` so condition is true and `OnInputClearClickAsync()` is called.
3. `ClearDefaultMode()` is called.
4. Value Equals to default(long?) so this method just returns and do nothing, SelectedOptionItems is not changed and user still see previous value.


- When I have special NULL value option in DataSource:
1. `Select.OnValueChange()` with NULL value is called.
2. `EvaluateValueChangedOutsideComponent()` is called.
3. ActiveOption is not NULL and Value Equals to default(long?) so this method just returns and do nothing, SelectedOptionItems is not changed and user still see previous value.

Or am I doing something wrong and missing something?

**Comment by anranruye** [comment]
@jeffraska Yes, I can reproduce this.
